fri/src/config.rs

```diff
@@ -1,18 +1,17 @@
 use core::marker::PhantomData;
 
-use p3_challenger::{CanObserve, GrindingChallenger};
+use p3_challenger::{CanObserve, CanSample, GrindingChallenger};
 use p3_commit::{DirectMmcs, Mmcs};
-use p3_field::{ExtensionField, PrimeField64, TwoAdicField};
+use p3_field::TwoAdicField;
 
 pub trait FriConfig {
```

Comment:
FriConfig only knows about `Challenge`, not `Val`. (except the `Witness` was a `Val`, so I made that its own associated type)

Reply:
Hm were there other reasons for removing `Val` from `FriConfig`, or it mainly comes down to what's most natural? To me it seems natural to keep `Val` in the config if our implementation deals with it in practice

Reply:
`fri::{prover::prove, verifier::verify}` never uses Val at all, except as a pow witness, not as a field. So it would be somewhat extraneous

Reply:
Since `FriConfig` mostly just contains `usize`s, maybe we take the trait bounds off of `FriConfig` and make it a simple 4-line struct, move some of the bounds to the actual functions, and do the big complicated trait/pimpl thing for `TwoAdicFriPcsConfig` / `TwoAdicFriPcs` instead..?

Reply:
I don't think it really matters tbh, this is just for two-adic performance, circle fri will probably be completely separate.